builtin/providers/google/resource_sql_database.go

```diff
@@ -67,6 +68,7 @@ func resourceSqlDatabaseCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	err = sqladminOperationWait(config, op, "Insert Database")
+	googleMutexKV.Unlock(instanceMutexKey(project, instance_name))
```

Comment:
Indeed, `defer` here will result in the `lock` persisting until `resourceSqlDatabaseRead` completes, so probably not what we want to do. 

- https://play.golang.org/p/61yvP1eLbs

That said, we probably do want to add an additional unlock in the error check block, yeah?